src/SDK/ConfigBuilder.php

```diff
@@ -0,0 +1,199 @@
+        'OTEL_EXPORTER_ZIPKIN_ENDPOINT' => 'trace.exporters.zipkin.endpoint',
+        'OTEL_EXPORTER_ZIPKIN_TIMEOUT' => 'trace.exporters.zipkin.timeout',
+        'OTEL_EXPORTER_OTLP_ENDPOINT' => 'trace.exporters.otlp.endpoint',
+        'OTEL_EXPORTER_OTLP_TRACES_ENDPOINT' => 'trace.exporters.otlp.endpoint',
+        'OTEL_EXPORTER_OTLP_INSECURE' => 'trace.exporters.otlp.insecure',
+        'OTEL_EXPORTER_OTLP_SPAN_INSECURE' => 'trace.exporters.otlp.insecure',
+        'OTEL_EXPORTER_OTLP_CERTIFICATE' => 'trace.exporters.otlp.certificate_file',
+        'OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE' => 'trace.exporters.otlp.certificate_file',
+        'OTEL_EXPORTER_OTLP_HEADERS' => 'trace.exporters.otlp.headers',
+        'OTEL_EXPORTER_OTLP_TRACES_HEADERS' => 'trace.exporters.otlp.headers',
+        'OTEL_EXPORTER_OTLP_COMPRESSION' => 'trace.exporters.otlp.compression',
+        'OTEL_EXPORTER_OTLP_TRACES_COMPRESSION' => 'trace.exporters.otlp.compression',
+        'OTEL_EXPORTER_OTLP_TIMEOUT' => 'trace.exporters.otlp.timeout',
+        'OTEL_EXPORTER_OTLP_TRACES_TIMEOUT' => 'trace.exporters.otlp.timeout',
+        'OTEL_EXPORTER_OTLP_PROTOCOL' => 'trace.exporters.otlp.protocol',
+        'OTEL_EXPORTER_OTLP_TRACES_PROTOCOL' => 'trace.exporters.otlp.protocol',
+        'OTEL_EXPORTER_JAEGER_AGENT_HOST' => 'trace.exporters.jaeger.agent_host',
+        'OTEL_EXPORTER_JAEGER_AGENT_PORT' => 'trace.exporters.jaeger.agent_port',
+        'OTEL_EXPORTER_JAEGER_ENDPOINT' => 'trace.exporters.jaeger.endpoint',
+        'OTEL_EXPORTER_JAEGER_TIMEOUT' => 'trace.exporters.jaeger.timeout',
+        'OTEL_EXPORTER_JAEGER_USER' => 'trace.exporters.jaeger.user',
+        'OTEL_EXPORTER_JAEGER_PASSWORD' => 'trace.exporters.jaeger.password',
+        'OTEL_PROPAGATORS' => 'propagators',
+        'OTEL_TRACES_EXPORTER' => 'trace.exporter',
+        'OTEL_METRICS_EXPORTER' => 'metrics.exporter',
+        'OTEL_PHP_TRACES_PROCESSOR' => 'trace.processor',
+    ];
+
+    public function build(array $data = []): object
+    {
+        //@phan-ignore PhanUndeclaredClassMethod
+        $schema = Expect::structure([
+            'log' => Expect::structure([
+                'stream' => Expect::string('php://stdout'),
+                'level' => Expect::string('info'),
+            ]),
+            'propagators' => Expect::string('tracecontext,baggage'),
+            'service' => Expect::structure([
+                'name' => Expect::string(),
+            ]),
+            'resource' => Expect::structure([
+                'limits' => Expect::structure([
+                    'attribute_count' => Expect::string()->castTo('int'),
+                    'attribute_value_length' => Expect::string()->castTo('int'),
+                ]),
+                'attributes' => Expect::array(),
+            ]),
+            'trace' => Expect::structure([
+                'sampler' => Expect::string('parentbased_always_on'),
```

Comment:
I found this method easier to reason about, because the config options for different types of sampler (or exporter/processor/etc) are easily found within their respective key in "samplers", where "sampler" is the list of samplers to be used.
So, iterate over "sampler" to work out which to create, and then look up sampler-specific config in samplers[type] as required.